// @vkontakte/vkjs/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    animate: function() {
        return _animate.animate;
    },
    animationEvent: function() {
        return _animate.animationEvent;
    },
    transitionEvent: function() {
        return _animate.transitionEvent;
    },
    waitAnimationEnd: function() {
        return _animate.waitAnimationEnd;
    },
    cancelWaitAnimationEnd: function() {
        return _animate.cancelWaitAnimationEnd;
    },
    waitTransitionEnd: function() {
        return _animate.waitTransitionEnd;
    },
    cancelWaitTransitionEnd: function() {
        return _animate.cancelWaitTransitionEnd;
    },
    createArray: function() {
        return _arrays.createArray;
    },
    sumArray: function() {
        return _arrays.sumArray;
    },
    averageArray: function() {
        return _arrays.averageArray;
    },
    uniqueArray: function() {
        return _arrays.uniqueArray;
    },
    shuffleArray: function() {
        return _arrays.shuffleArray;
    },
    chunkArray: function() {
        return _arrays.chunkArray;
    },
    omitFromArray: function() {
        return _arrays.omitFromArray;
    },
    difference: function() {
        return _arrays.difference;
    },
    /**
 * classNames
 */ classNames: function() {
        return _classNames.classNames;
    },
    /**
 * clipboard
 */ copyTextToClipboard: function() {
        return _clipboard.copyTextToClipboard;
    },
    /**
 * common
 */ isNumeric: function() {
        return _common.isNumeric;
    },
    /**
 * detections
 */ isPassiveEventsSupported: function() {
        return _detections.isPassiveEventsSupported;
    },
    isSmoothScrollSupported: function() {
        return _detections.isSmoothScrollSupported;
    },
    isEqual: function() {
        return _equal.isEqual;
    },
    noop: function() {
        return _functions.noop;
    },
    throttle: function() {
        return _functions.throttle;
    },
    debounce: function() {
        return _functions.debounce;
    },
    once: function() {
        return _functions.once;
    },
    asyncImportLoader: function() {
        return _async.asyncImportLoader;
    },
    getCookie: function() {
        return _cookie.getCookie;
    },
    isCookieEnabled: function() {
        return _cookie.isCookieEnabled;
    },
    SECONDS_IN_THE_DAY: function() {
        return _date.SECONDS_IN_THE_DAY;
    },
    isDateToday: function() {
        return _date.isDateToday;
    },
    isDateYesterday: function() {
        return _date.isDateYesterday;
    },
    isDateTomorrow: function() {
        return _date.isDateTomorrow;
    },
    isSameDate: function() {
        return _date.isSameDate;
    },
    getBeginningOfDay: function() {
        return _date.getBeginningOfDay;
    },
    isLeapYear: function() {
        return _date.isLeapYear;
    },
    getLastDayOfMonth: function() {
        return _date.getLastDayOfMonth;
    },
    getStartOfWeek: function() {
        return _date.getStartOfWeek;
    },
    addDays: function() {
        return _date.addDays;
    },
    createDateFromUnixTimestamp: function() {
        return _date.createDateFromUnixTimestamp;
    },
    getUnixTimestampFromDate: function() {
        return _date.getUnixTimestampFromDate;
    },
    convertDateToInputFormat: function() {
        return _date.convertDateToInputFormat;
    },
    formatDuration: function() {
        return _duration.formatDuration;
    },
    /**
 * OffsetRect
 */ getOffsetRect: function() {
        return _getOffsetRect.getOffsetRect;
    },
    getPhotoSize: function() {
        return _getPhotoSize.getPhotoSize;
    },
    escape: function() {
        return _escape.escape;
    },
    unescape: function() {
        return _escape.unescape;
    },
    encodeHTMLEntities: function() {
        return _escape.encodeHTMLEntities;
    },
    decodeHTMLEntities: function() {
        return _escape.decodeHTMLEntities;
    },
    decodeHTMLEntitiesDeep: function() {
        return _escape.decodeHTMLEntitiesDeep;
    },
    decodeHTMLFullEntities: function() {
        return _escape.decodeHTMLFullEntities;
    },
    leadingZero: function() {
        return _numbers.leadingZero;
    },
    formatNumber: function() {
        return _numbers.formatNumber;
    },
    /**
 * objects
 */ deleteObjectKeys: function() {
        return _objects.deleteObjectKeys;
    },
    /**
 * querystring
 */ querystring: function() {
        return _querystring.querystring;
    },
    /**
 * random
 */ getRandomInt: function() {
        return _random.getRandomInt;
    },
    getRandomString: function() {
        return _random.getRandomString;
    },
    hasReactNode: function() {
        return _react_utils.hasReactNode;
    },
    isPrimitiveReactNode: function() {
        return _react_utils.isPrimitiveReactNode;
    },
    isObjectLike: function() {
        return _type_checkers.isObjectLike;
    },
    isArray: function() {
        return _type_checkers.isArray;
    },
    isObject: function() {
        return _type_checkers.isObject;
    },
    isUndefined: function() {
        return _type_checkers.isUndefined;
    },
    isFunction: function() {
        return _type_checkers.isFunction;
    },
    isFormData: function() {
        return _type_checkers.isFormData;
    },
    isString: function() {
        return _type_checkers.isString;
    },
    isNumber: function() {
        return _type_checkers.isNumber;
    },
    escapeRegExp: function() {
        return _regexp.escapeRegExp;
    },
    isRetina: function() {
        return _retina.isRetina;
    },
    localStorage: function() {
        return _storage.localStorage;
    },
    sessionStorage: function() {
        return _storage.sessionStorage;
    },
    hasMouse: function() {
        return _InputUtils.hasMouse;
    },
    hasHover: function() {
        return _InputUtils.hasHover;
    },
    hasTouchEvents: function() {
        return _InputUtils.hasTouchEvents;
    },
    hasTouch: function() {
        return _InputUtils.hasTouch;
    },
    canUseDOM: function() {
        return _dom.canUseDOM;
    },
    canUseEventListeners: function() {
        return _dom.canUseEventListeners;
    },
    onDOMLoaded: function() {
        return _dom.onDOMLoaded;
    },
    IPHONE_SAFARI_BOTTOM_BAR: function() {
        return _IOSDetections.IPHONE_SAFARI_BOTTOM_BAR;
    },
    IPHONE_X_SAFARI_BOTTOM_BAR: function() {
        return _IOSDetections.IPHONE_X_SAFARI_BOTTOM_BAR;
    },
    IPHONE_KEYBOARD_REJECT_OFFSET: function() {
        return _IOSDetections.IPHONE_KEYBOARD_REJECT_OFFSET;
    },
    IOS_NO_KEYBOARD_ALLOWED_OFFSET: function() {
        return _IOSDetections.IOS_NO_KEYBOARD_ALLOWED_OFFSET;
    },
    detectIOS: function() {
        return _IOSDetections.detectIOS;
    },
    isIPad: function() {
        return _IOSDetections.isIPad;
    },
    isIPhone: function() {
        return _IOSDetections.isIPhone;
    },
    isIOS: function() {
        return _IOSDetections.isIOS;
    },
    isIPadOS: function() {
        return _IOSDetections.isIPadOS;
    },
    iosMajor: function() {
        return _IOSDetections.iosMajor;
    },
    iosMinor: function() {
        return _IOSDetections.iosMinor;
    },
    isWKWebView: function() {
        return _IOSDetections.isWKWebView;
    },
    isScrollBasedViewport: function() {
        return _IOSDetections.isScrollBasedViewport;
    },
    isIPhoneX: function() {
        return _IOSDetections.isIPhoneX;
    },
    isIOSChrome: function() {
        return _IOSDetections.isIOSChrome;
    },
    isLandscapePhone: function() {
        return _IOSDetections.isLandscapePhone;
    },
    checkIPadOS: function() {
        return _IOSDetections.checkIPadOS;
    }
});
var _export_star = require("@swc/helpers/_/_export_star");
var _animate = require("./animate");
var _arrays = require("./arrays");
var _classNames = require("./classNames");
var _clipboard = require("./clipboard");
var _common = require("./common");
var _detections = require("./detections");
var _equal = require("./equal");
var _functions = require("./functions");
var _async = require("./async");
var _cookie = require("./cookie");
var _date = require("./date");
var _duration = require("./duration");
var _getOffsetRect = require("./getOffsetRect");
var _getPhotoSize = require("./getPhotoSize");
var _escape = require("./html/escape");
var _numbers = require("./numbers");
var _objects = require("./objects");
var _querystring = require("./querystring");
var _random = require("./random");
var _react_utils = require("./react_utils");
var _type_checkers = require("./type_checkers");
var _regexp = require("./regexp");
var _retina = require("./retina");
var _storage = require("./storage");
var _InputUtils = require("./InputUtils");
var _dom = require("./dom");
var _IOSDetections = require("./IOSDetections");
_export_star._(require("./text"), exports);
